lib/Service/Gateway/SMS/Provider/PuzzelSMS.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Kim Syversen <kim.syversen@gmail.com>
+ *
+ * Nextcloud - Two-factor Gateway
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\TwoFactorGateway\Service\Gateway\SMS\Provider;
+
+use Exception;
+use OCA\TwoFactorGateway\Exception\SmsTransmissionException;
+use OCP\Http\Client\IClient;
+use OCP\Http\Client\IClientService;
+use OCP\IConfig;
+
+class PuzzelSMS implements IProvider {
+
+	const PROVIDER_ID = 'puzzelsms';
+
+	/** @var IClient */
+	private $client;
+
+	/** @var PuzzelConfig */
+	private $config;
+
+	public function __construct(IClientService $clientService,
+								PuzzelSMSConfig $config) {
+		$this->client = $clientService->newClient();
+		$this->config = $config;
+	}
+
+	/**
+	 * @param string $identifier
+	 * @param string $message
+	 *
+	 * @throws SmsTransmissionException
+	 */
+	public function send(string $identifier, string $message) {
+		$config = $this->getConfig();
+
+		try {
+			$this->client->get(
+				$config->getUrl(),
+				[
+					'query' => [
+						'username' => $config->getUser(),
+						'password' => $config->getPassword(),
+						"message[0].recipient" => "+".$identifier,
```

Comment:
Not sure if we should add the `+` here or let users enter it in that notation

Reply:
That part of the code has been used in production with the earlier twofactor_sms-version of the app for about 1 year now. 

The `+` was added due to users having issues with not receiving any SMS and it have worked flawlessly since then. Because of that I would say it should be there.